calendar-app/src/pages/calendar/MainContent/TimeframeToday.tsx

```diff
@@ -0,0 +1,55 @@
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../../../store";
+import { useEffect } from "react";
+import { setEvents } from "../../../features/eventDisplay";
+import { retrieveEventsFromServer } from "../../../api/getEvents";
+import { StoredEvent } from "../../../utils/types";
+export default function TimeframeToday() {
+  const dispatch = useDispatch();
+  const currentDateStr = useSelector(
+    (state: RootState) => state.currentDate.currentDate
+  );
+  useEffect(() => {
+    const currentDate = new Date(currentDateStr);
+
+    const weekKey = getWeekKey(currentDate);
+
+    const cached = sessionStorage.getItem(weekKey);
+    if (cached) {
+      dispatch(setEvents(JSON.parse(cached)));
+      return;
+    }
```

Comment:
it would be better to handle such state initialisation somewhere at the store's or app initialisation level, not inside components effects